Add new project

davidmatei opened the pull request:
-aanpassingen CreateProject : 
 * Module selecteren
 * User definieren
 * Hard coded date weggewerkt


-EditProject:
* Edit project 
*Vieuws

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

# Conflicts:
#	DAL/Seed.cs
#	DAL/repos/IdeationQuestionsRepository.cs
#	UI-MVC/Views/Shared/_CookieConsentPartial.cshtml

davidmatei committed:
projectfuncties

1e versie projectfuncties

davidmatei committed:
begin van projectcontroller. Vandaag alles ervan afwerken

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

davidmatei committed:
Merge branch 'dev' into add-new-project

# Conflicts:
#	DB-Scripts/IP1_TSI_CREATE_SCRIPT.sql
#	DB-Scripts/IP1_TSI_SEED_SCRIPT.sql
#	Domain/UserInput/MultipleAnswer.cs
#	Domain/Users/UserDetail.cs
#	UI-MVC/Areas/Identity/Data/UIMVCUser.cs
#	UI-MVC/Areas/Identity/Pages/Account/Login.cshtml
#	UI-MVC/Areas/Identity/Pages/Account/Register.cshtml
#	UI-MVC/Areas/Identity/Pages/Account/Register.cshtml.cs
#	UI-MVC/Migrations/UIMVCContextModelSnapshot.cs
#	UI-MVC/ScaffoldingReadme.txt
#	UI-MVC/UIMVC.csproj
#	UI-MVC/UIMVC.db

davidmatei committed:
-create project en 1 phase
-tonen van project detail met bijhorende phases
-project manager refactoren: phases van project ophalen en populaten

davidmatei committed:
-meerdere phases in project toevoegen

davidmatei committed:
-create project en 1 phase
-tonen van project detail met bijhorende phases
-project manager refactoren: phases van project ophalen en populaten

davidmatei committed:
-create project en 1 phase
-tonen van project detail met bijhorende phases
-project manager refactoren: phases van project ophalen en populaten

davidmatei committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	Domain/Projects/Project.cs

davidmatei committed:
-dev mergen

davidmatei committed:
Merge branch 'dev' into add-new-project

# Conflicts:
#	BL/ProjectManager.cs
#	UI-MVC/Services/ProjectService.cs
#	UI-MVC/TheSpanishDatabase.db
#	UI-MVC/wwwroot/dist/site.entry.js.map

davidmatei committed:
-Phases bijgewerkt

davidmatei committed:
update

-git fix

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

davidmatei committed:
Merge branch 'dev' into add-new-project

davidmatei committed:
createProject en EditProject

- CreateProject
- EditProject

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

davidmatei committed:
Merge branch 'dev' into add-new-project

NielsSensei commented on BL/ProjectManager.cs:
```diff
@@ -43,15 +46,37 @@ public void EditProject(Project project)
+            return project;
+        }
+
+        public IEnumerable<Project> GetProjects(int projectId)
```
Waarom roept de service hier de GetProject(int projectId, bool details) niet aan. Deze methode wordt ook niet gebruikt buiten de service die er geen nut van heeft.

NielsSensei commented on BL/ProjectManager.cs:
```diff
@@ -43,15 +46,37 @@ public void EditProject(Project project)
+            return projects;
+        }
+
+        public IEnumerable<Project> GetProjects()
```
De ProjectService heeft een gelijkaardige methode die gebruik maakt van dependency injection. see @CollectPlatformProjects(Platform platform)

NielsSensei commented on BL/ProjectManager.cs:
```diff
@@ -114,7 +139,10 @@ public void RemovePhase(int projectId, int phaseId)
             }
-            ProjectRepo.Delete(phaseId);
+
+            //TODO: Enkel de phase moet verwijdert worden project niet? 
```
Inderdaad. Voor IPT13-28 (Niet voor deze feature) is het best dat je deze methode herschrijft.

NielsSensei commented on Domain/Projects/Phase.cs:
```diff
@@ -4,14 +4,21 @@ namespace Domain.Projects
         public DateTime StartDate { get; set; }
         public DateTime EndDate { get; set; }
 
+        public bool IsCurrentPhase { get; set; }
```
Ik snap het nut van deze property maar is dit de responsibility van het domein om dit bij te houden. Zo ja moet deze boolean worden gepersisteerd?

NielsSensei commented on Domain/Projects/Project.cs:
```diff
@@ -6,16 +7,40 @@
+       
         public string Goal { get; set; }
-        public string Status { get; set; }
+        public string Status { get; set; } //TODO bool? 
```
Reeds besproken: bool is niet modulair. Zowel enum als string (zoals wij hier doen bvb. OPEN, GESLOTEN, NIET ZICHTBAAR, ...)

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,178 @@
+
+        #region Index
+
+        public ActionResult Index()
```
Doet hetzelfde als localhost:5000/Platform/1 (regardless of welke index als er een platform voor bestaat)

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,178 @@
+            return View(projects);
+        }
+
+        public IActionResult Details(int id)
```
Doet hetzelfde als localhost:5000/Platform/CollectProject/1 (maakt niet uit welke id zolang ie bestaat)

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,178 @@
+        //GET: /Project/Detail/<project_id>
+
+
+        public IActionResult Delete(object id)
```
Is IPT13-28 (hoort niet bij deze feature)

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,178 @@
+
+        [HttpPost]
+        //[Authorize]
+        public IActionResult Edit(Project project)
```
IPT13-26 (niet deze feature) maar andere feature

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,178 @@
+        [HttpPost]
+        public IActionResult Create(ProjectViewModel pvm, string user)
+        {
+            UIMVCUser projectUser = _userManager.GetUserAsync(HttpContext.User).Result;
```
Hij stuurt de user niet juist mee. Ik stel voor dat bij aanroep van de Create (de HttpGet) Dat je de string userId meestuurt.

NielsSensei committed:
Dit gecomment door ouwe merge te fixen | Niet meer nodig. Opgelost in ouwe feature

NielsSensei commented on UI-MVC/Views/Platform/Index.cshtml:
```diff
@@ -22,7 +22,8 @@
                  <div class="col-md-3">
                      @Html.DisplayFor(m => project)
-                     <a asp-controller="Platform" asp-action="CollectProject" asp-route-id="@project.Id">Ga naar Project</a>                
+                     <a asp-controller="Platform" asp-action="CollectProject" asp-route-id="@project.Id">Ga naar Project</a>  
```
Deze view is deprecated

NielsSensei commented on UI-MVC/Views/Project/Create.cshtml:
```diff
@@ -0,0 +1,94 @@
+            <input type="date" asp-for="Project.EndDate" class="form-control" value="1/1/0001 12:00:00 AM"/>
+        </div>
+        <div class="form-group col-md-6">
+            <label asp-for="Project.Goal">Hoofdvraag</label>
```
Ik was eerst niet zeker over wat je het hier had, bleek doel te zijn. Hernoem dit maakt het duidelijker.

NielsSensei commented on UI-MVC/Views/Project/Create.cshtml:
```diff
@@ -0,0 +1,94 @@
+            <input asp-for="Project.Goal" class="form-control" value="Test"/>
+        </div>
+        <div class="form-group col-md-6">
+            <label asp-for="Project.Status">Status</label>
```
Ofwel status niet editable maken oftewel ervoor zorgen dat deze automatisch in uppercase naar de manager gaat.

NielsSensei commented on UI-MVC/Views/Project/Create.cshtml:
```diff
@@ -0,0 +1,94 @@
+
+    <div class="form-group">
+
+        <label asp-for="Project.Platform.Name">Platform Naam</label>
```
Als je de button verplaats kan je in de asp-route een platform id meegeven waardoor je geen platform meer 
moet kiezen (wat je nu doet is bad practice want je haalt miss tientallen platformen op ipv 1 idke op te 
sturen)

NielsSensei commented on UI-MVC/Views/Project/Details.cshtml:
```diff
@@ -0,0 +1,28 @@
+
+
+@{
+    ViewBag.Title = "Project";
```
View is deprecated

NielsSensei commented on UI-MVC/Views/Project/Index.cshtml:
```diff
@@ -0,0 +1,42 @@
+@using UIMVC.Services
+@inject ProjectService ProjectService
+@{
+    ViewBag.Title = "Index";
```
View is deprecated

NielsSensei commented on UI-MVC/Views/Shared/_ProjectPhaseDetailPartial.cshtml:
```diff
@@ -0,0 +1,8 @@
+@model Domain.Projects.Phase
+
+<dl>
```
Deprecated view

NielsSensei reviewed (dismissed):
Algemeen:
Button voor create project op site/Platform/index
Als je voor textboxen al een value wilt gebruik de attribute placeholder
Je mag alles buiten de code (achter de motorkap) in het Nederlands doen.
Maak gebruik van code conventions
![image](https://user-images.githubusercontent.com/44059248/57168295-7ba5d900-6e01-11e9-9a97-2b5864771928.png)


Dingen getest (na en tijdens alle code overlopen):
Maak een Platform Naast de opmerkingen is de code dik ok.
Edit: nog niet de feature maar wel gewoon al aangemaakt, cva.
Delete: nog niet de feature maar wel gewoon al aangemaakt, cva.
Details: Deprecated

Other:
Ik zal dat van die phases wel doen met javascript (ik moet hier en daar toch wat js toepassen)

davidmatei committed:
-In ProjectService, methode Getprojects verwijdert.
-Project Views: Index en Detail verwijdert en vervangen door de Index van Platform
-Button in localhost:5000/Platform/. om een project aan te maken. Platform id wordt meegegeven
-Project Create:
*de platform wordt meegegeven
*bool isCurrentPhase weggedaan, currentphase wordt ingevuld in de ProjectViewModel

davidmatei committed:
Merge remote-tracking branch 'origin/add-new-project' into add-new-project

NielsSensei committed:
Merge branch 'dev' into add-new-project

NielsSensei committed:
small fix

NielsSensei reviewed (changes requested):
Doel wordt niet meegestuurd:
![image](https://user-images.githubusercontent.com/44059248/57233403-db2d0000-701e-11e9-8ece-78a1d35ca92d.png)

Nog altijd geen gebruik van placeholders x2.

Er wordt geen fase aangemaakt bij waardoor je niet het project kan tonen/zien.

Nog altijd geen conventions x2
![image](https://user-images.githubusercontent.com/44059248/57233468-fdbf1900-701e-11e9-9765-d0a4f14f3d99.png)

Nog altijd half engels, half nederlands op de view. x2

De rest van de commentaar zal in de code zitten indien van toepassing.

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,166 @@
+
+        [Authorize]
+        [HttpGet]
+        public IActionResult Create(int platform)
```
conventions geval 1:

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,166 @@
+
+        [Authorize]
+        [HttpPost]
+        public IActionResult Create(ProjectViewModel pvm, int platform)
```
conventions geval 2:

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,166 @@
+            Project pr = new Project()
+            {
+                User = projectUser,
+                CurrentPhase = pvm.Project.CurrentPhase,
```
currentphase wordt niet gevuld de id in de db is 0.

davidmatei replied:
verder in de code staat er een foreach loop = daar wordt fases aangepakt in de databank en wordt de id meegegeven en gekoppelt

NielsSensei replied:
ik spreek hier niet over de hele lijst. Ik spreek hier over de currentphase.

![image](https://user-images.githubusercontent.com/44059248/57237070-b177d700-7026-11e9-81b6-3321a3abf414.png)

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,166 @@
+
+        //[Authorize]
+        [HttpGet]
+        public IActionResult Edit(int id)
```
code conventions nr 3:

NielsSensei commented on UI-MVC/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,166 @@
+
+        [HttpPost]
+        //[Authorize]
+        public IActionResult Edit(Project project)
```
code convention nr 4:

NielsSensei commented on UI-MVC/Models/ProjectViewModel.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public class ProjectViewModel
+    {
+        public Project Project { get; set; }
```
zoals besproken hier een currentphase erbij of ProjectViewModel.Project.Currentphase vullen.

NielsSensei commented on UI-MVC/Views/Platform/Index.cshtml:
```diff
@@ -11,20 +11,24 @@
     <hr/>
     <p>This is where I'd put an image if I (knew how I) could convert from BLOB</p>
     <hr/>
+    <a asp-controller="Project" asp-action="Create" asp-route-platform="@Model.Id">Maak nieuw project aan</a>
```
code convention nr 5/6:

NielsSensei commented on UI-MVC/Views/Platform/Index.cshtml:
```diff
@@ -11,20 +11,24 @@
+                <div class="col-md-3">
+                    @Html.DisplayFor(m => project)
+                    <a asp-controller="Platform" asp-action="CollectProject" asp-route-id="@project.Id">Ga naar Project</a>
+                    <a asp-controller="Project" asp-action="Edit" asp-route-id="@project.Id">Pas project aan</a>
```
code convention nr 6/7 (lost count):

NielsSensei commented on UI-MVC/Views/Project/Create.cshtml:
```diff
@@ -0,0 +1,89 @@
+}
+
+
+<h2>Create Project</h2>
```
code convention nr 8/9 (Heel deze page)

NielsSensei commented on UI-MVC/Views/Project/Edit.cshtml:
```diff
@@ -0,0 +1,69 @@
+@model Domain.Projects.Project
+
+@{
+    ViewBag.Title = "Edit: " + Model.Title;
```
code convention nr 9/10: (heel deze page)

davidmatei committed:
-CurrentFase aanduiden door gebruik van bool IsCurrent => wordt gebruikt om de currentFase te kunnen aanduiden,
- MakeProject aangepast => maakt phases hierin aan en update de project telkens op na aanmaak van phases

davidmatei committed:
-Phases bijgewerkt

davidmatei committed:
Premerge

davidmatei committed:
Merge branch 'dev' of https://github.com/NielsSensei/TreeCompany_TheSpanishInquisition_IP1_Team13_KdG1819 into dev

davidmatei committed:
Merge branch 'dev' into add-new-project

NielsSensei commented:
Zoals besproken, David heeft andere branch op basis van deze waar hij ook delete/edit doet. Hij had voor de lunch dit normaal klaar gekregen en ik verwacht soon een nieuwe PR.
